src/Core/Grid/Action/Row/AccessibilityChecker/AccessibilityCheckerInterface.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Action\Row\AccessibilityChecker;
+
+/**
+ * Interface ApplicabilityCheckerInterface.
+ */
+interface AccessibilityCheckerInterface
```

Comment:
@mickaelandrieu once you have time, i need your input on this one. Its the same idea we have talked about a few days ago: by default all defined row actions are available, however, if you wish yo have some access check you may define your own `AccessabilityChecker` and set it as option on action, for example:

```php
            ->add((new LinkRowAction('view'))
                ->setOptions([
                    'route' => 'admin_category_listing',
                    'route_param_name' => 'id_category',
                    'route_param_field' => 'id_category',
                    'accessibility_checker' => $this->categoryForViewAccessibilityChecker,
                ])
            )
```
im still thinking if `ApplicabilityCheckerInterface::isApplicable()` would be a better name for it?

Reply:
First of all, good job to have found a "not that bad" workaround".

We need to work on something that is "included" in the Grid: we may define the dos and donts of this feature next week and make it part of 1.7.6, don't you think?

This could remove a "bug" from 1.6 I have found: be able to click on an action link (like "edit/see/create") when we already know the logged user have no rights to do it.

So we have this subject that could be applied to every grid, and some specific use cases like the one we see here.